### PR TITLE
[stable9] minor smb logging, do not use deprecated method

### PR DESCRIPTION
**Pull request by butonic**
add log for the actual smb implementation that is used (libsmbclient vs smbclient)

already contained in master and stable9 PRs

cc @felixboehm @DeepDiver1975 @PVince81

### CODE REVIEW

File: apps/files_external/lib/smb.php [hunk 1]
```diff
@@ -410,7 +412,7 @@ public function fopen($path, $mode) {
 						if (!$this->isCreatable(dirname($path))) {
 							break;
 						}
-						$tmpFile = \OCP\Files::tmpFile($ext);
+						$tmpFile = \OC::$server->getTempManager()->getTemporaryFile($ext);
```
Comment (VicDeo) [inline apps/files_external/lib/smb.php]: @butonic  interrresting. Why we are creating file with the name which is either `strrpos($path, '.')` or empty string in the first place?

Comment (VicDeo) [inline apps/files_external/lib/smb.php]: and what happens if any directory in `$path` has a dot: `/something/that/is/not.supposed/tohappen`

Comment (butonic) [inline apps/files_external/lib/smb.php]: um from phpdoc `* @param string $postFix Postfix appended to the temporary file name`. so $ext is just appended to have the same filetype if it is given. That being said `/something/that/is/not.supposed/tohappen` is a valid point. I guess
- [ ] dropping the $ext is fine.
- [ ] add a PR for a unit test that extends directory providers with `/something/that/is/not.supposed/tohappen`

@VicDeo good catch!

**Comment by PVince81** [comment]
any update ? would move to 9.0.6

**Comment by butonic** [comment]
tests are coming out green. @PVince81 @VicDeo @jvillafanez

**Comment by butonic** [comment]
I take [that](https://github.com/owncloud/core/pull/25994#issuecomment-247958141) back:
```

12:29:14 1) Test\Files\Storage\DAV::testWriteWithDotInPath
12:29:14 OCP\Files\StorageNotAvailableException: Sabre\HTTP\ClientHttpException: Conflict
12:29:14 
12:29:14 /var/lib/jenkins/workspace/owncloud-core/core/PR-25994/lib/private/files/storage/dav.php:797
12:29:14 /var/lib/jenkins/workspace/owncloud-core/core/PR-25994/lib/private/files/storage/dav.php:638
12:29:14 /var/lib/jenkins/workspace/owncloud-core/core/PR-25994/lib/private/files/storage/dav.php:177
12:29:14 /var/lib/jenkins/workspace/owncloud-core/core/PR-25994/tests/lib/files/storage/storage.php:152

```

investigating

**Comment by jvillafanez** [comment]
Changes look good expect for the failing test.

**Comment by PVince81** [comment]
Moving to 9.0.7.

**Comment by PVince81** [comment]
@butonic any update ?

**Comment by PVince81** [comment]
is this backport still worth it ?

**Comment by PVince81** [comment]
no feedback since sep 2016, closing

**Pull request closed without merging**